polyfill/lib/intl.mjs

```diff
@@ -319,15 +358,39 @@ function extractOverrides(temporalObj, main) {
       );
     }
     return {
-      instant: main[TIMEZONE].getInstantFor(datetime),
+      instant: main[TZ_RESOLVED].getInstantFor(datetime),
       formatter: main[DATETIME]
     };
   }
 
+  if (ES.IsTemporalZonedDateTime(temporalObj)) {
+    const calendar = GetSlot(temporalObj, CALENDAR);
+    if (calendar.id !== 'iso8601' && calendar.id !== main[CAL_ID]) {
+      throw new RangeError(
+        `cannot format ZonedDateTime with calendar ${calendar.id} in locale with calendar ${main[CAL_ID]}`
```

Comment:
Use `CalendarToString` (and make sure to call it just once.)

Reply:
Good point, I've added another commit to fix the existing instances of this problem as well.